src/Framework/Receipts/Actions/GenerateConfirmationPageReceipt.php

```diff
@@ -148,6 +151,26 @@ private function fillDonationDetails(DonationReceipt $receipt)
             );
         }
 
+        $eventTickets = give()->eventTickets->queryByDonationId($receipt->donation->id)->getAll();
+
+        if ( ! empty($eventTickets)) {
+            $currency = $receipt->donation->amount->getCurrency();
+            $totalAmountValue = array_reduce($eventTickets, function ($acc, $eventTicket) {
+                $ticketType = $eventTicket->ticketType()->get();
+
+                return $acc + $ticketType->price->getAmount();
```

Comment:
The underlying `Money` library has methods for calculating and combining multiple values.

Instead of initializing the `array_reduce` with `0` we can initialize it with a `Money` object of zero value and then add the amounts to that.

```php
$total = array_reduce($eventTickets, function(Money $carry, EventTicket $eventTicket) {
    return $carry->add(
        $event->ticketType->price
    );
}, new Money(0, $currency));
```

Reply:
Refactored here: d751596339aec1d5868e3124696ade097f811f56